File: Backend/src/main/java/com/food/service/DeliveryService.java
package com.food.service;

import java.util.List;

import com.food.DTO.DeliveryDTO;
import com.food.DTO.DeliveryResponseDTO;

public interface DeliveryService {
	// Create delivery
	public String createDelivery(DeliveryDTO request);

	//Find by id
	public DeliveryResponseDTO findById(Long id);

	//find the assigned delivery (for the CURRENTLY LOGGED IN volunteer)
	public List<DeliveryResponseDTO> findAssignedDeliveries();

	//start delivery
	public String startDelivery(Long id);

	//complete delivery
	public String completeDelivery(Long id);

	//track delivery
	public DeliveryResponseDTO trackDelivery(Long id);
}

File: FoodProject_Frontend/ProjectFood/src/services/deliveryService.js
import api from "./api";

// Deliveries
export const createDelivery = (data) => api.post("/deliveries", data);
export const getDeliveryById = (id) => api.get(`/deliveries/${id}`);
export const getAssignedDeliveries = () => api.get("/deliveries/assigned");
export const startDelivery = (id) => api.put(`/deliveries/${id}/start`);
export const completeDelivery = (id) => api.put(`/deliveries/${id}/complete`);
export const trackDelivery = (id) => api.get(`/deliveries/${id}/track`);

export default {
    createDelivery,
    getDeliveryById,
    getAssignedDeliveries,
    startDelivery,
    completeDelivery,
    trackDelivery,
};
